src/rockstor/scripts/scheduled_tasks/reboot_shutdown.py

```diff
@@ -33,6 +38,45 @@ def validate_shutdown_meta(meta):
     return meta
 
 
+def all_devices_offline(addresses):
+    for address in addresses:
+        logger.debug('Pinging ' + address + '...')
+        if is_network_device_responding(address):
+            return False
+    return True
+
+
+def run_conditions_met(meta):
+    if meta['ping_scan']:
+        address_parser = csv_reader([meta['ping_scan_addresses']], delimiter=',')
+        addresses = list(address_parser)
+
+        if len(addresses) > 0:
+            addresses = addresses[0]
+            # remove all non word characters except from '.' and '-' to prevent
+            # the execution of shell code when running the ping command
+            addresses = [re.sub(r"[^\w\.-]", '', address) for address in addresses]
+            interval = float(meta['ping_scan_interval']) if float(meta['ping_scan_interval']) >= 5 else 5.0
+            iterations = int(meta['ping_scan_iterations']) if int(meta['ping_scan_iterations']) >= 1 else 1
+
+            logger.debug('Pinging devices: '+str(addresses))
```

Comment:
Small style reformatting:
```suggestion
            logger.debug('Pinging devices: {}'.format(addresses))
```